components/micro-gateway-core/src/main/java/org/wso2/micro/gateway/core/globalthrottle/databridge/agent/endpoint/DataEndpointGroup.java

```diff
@@ -411,14 +434,13 @@ private boolean isServerExists(DataEndpoint dataEndpoint, String protocol, Strin
                     return true;
                 } else {
                     // this block is executed when connection is SSL
-                    DataEndpointConfiguration dataEndpointConfiguration = dataEndpoint.getDataEndpointConfiguration();
-                    Socket socket = (Socket) dataEndpointConfiguration.getSecuredTransportPool().borrowObject(
-                            dataEndpointConfiguration.getPublisherKey());
+                    SSLSocket socket = (SSLSocket) sslSocketFactory.createSocket(ip, port);
                     OutputStream outputStream = socket.getOutputStream();
-                    String sessionId = dataEndpointConfiguration.getSessionId();
+                    String sessionId = dataEndpoint.getDataEndpointConfiguration().getSessionId();
                     ByteBuffer buf = ByteBuffer.allocate(sessionId.length());
                     outputStream.write(buf.array());
                     outputStream.flush();
+                    socket.close();
```

Comment:
Use try finally here

Reply:
resolved in https://github.com/wso2/product-microgateway/pull/2579/commits/4aee1212c5e18f2e348c4f6de6855fff3eec5039